lib/validation.rb

```diff
@@ -0,0 +1,30 @@
+class Validation
+  attr_reader :valid
+  def initialize
+    @valid = false
+  end
+
+  def ask_sign(sign)
+    until @valid
+      print "Please enter 1 character for the sign\nPlease enter another sign: " if sign.length > 1
+      sign = gets.chomp if sign.length > 1
+      next unless sign.length == 1
+
+      @valid = true
+    end
+    @valid = false
+    sign
+  end
+
+  def ask_name(name1, name2)
+    until @valid
+      print "#{name1} is already taken\nPlease enter another name: " if name1 == name2
+      name2 = gets.chomp if name1 == name2
+      next unless name1 != name2
+
+      @valid = true
```

Comment:
- [ ] Kindly remove all print statements here and let the `main.rb` handle them as I have stated in my earlier comment. Possible ways to go about this is to assign the string to a variable and make the method return it OR you make your methods accept some parameters and based on the action from a player, you can output a message.

Reply:
If we move all these prints, we also have to move the validation methods, and some crucial methods from the classes, which will make the main.rb look messy. So we did,t do that. But we are actually printing everything by calling a print method from main.rb.